torch/fx/_pytree.py

```diff
@@ -0,0 +1,34 @@
+from typing import Callable, Any, Tuple, List, Dict, Type
+from torch.utils._pytree import PyTree, TreeSpec, LeafSpec
+
+FlattenFuncSpec = Callable[[PyTree, TreeSpec], List]
+
+SUPPORTED_NODES: Dict[Type[Any], Any] = {}
+def _register_pytree_flatten_spec(typ: Any, flatten_fn_spec: FlattenFuncSpec) -> None:
+    SUPPORTED_NODES[typ] = flatten_fn_spec
+
+def tree_flatten_spec(pytree: PyTree, spec: TreeSpec) -> List[Any]:
+    if isinstance(spec, LeafSpec):
+        return [pytree]
+    flatten_fn_spec = SUPPORTED_NODES[spec.type]
+    if flatten_fn_spec is None:
+        raise RuntimeError(f"Cannot flatten pytree {type(pytree)} from spec")
```

Comment:
Is this a user-actionable error or would it indicate an issue in our code? If it's user visible, can we add some more descriptive messaging about what to do to fix the issue?

Reply:
This is an user-actionable error - it indicates that they've passed in a different pytree from the one they originally flattened the function with.

I'll modify the error to provide the expected and given specs.